src/components/dragLineLayer.ts

```diff
@@ -0,0 +1,223 @@
+
+export module Components {
+  export class DragLineLayer<D> extends GuideLineLayer<D> {
+    private _dragInteraction: Interactions.Drag;
+    private _detectionRadius = 3;
+    private _detectionEdge: d3.Selection<void>;
+    private _enabled = true;
+
+    private _dragStartCallbacks: Utils.CallbackSet<DragLineCallback<D>>;
+    private _dragCallbacks: Utils.CallbackSet<DragLineCallback<D>>;
+    private _dragEndCallbacks: Utils.CallbackSet<DragLineCallback<D>>;
+    private _disconnectInteraction: () => void;
+
+    constructor(orientation: string) {
+      super(orientation);
+      this.addClass("drag-line-layer");
+      this.addClass("enabled");
+
+      this._dragInteraction = new Plottable.Interactions.Drag();
+      this._dragInteraction.attachTo(this);
+
+      let onLine = (p: Point) => {
+        return (this._isVertical() &&
+                 this.pixelPosition() - this.detectionRadius() <= p.x &&
+                 p.x <= this.pixelPosition() + this.detectionRadius()
+               ) ||
+               (
+                 !this._isVertical() &&
+                 this.pixelPosition() - this.detectionRadius() <= p.y &&
+                 p.y <= this.pixelPosition() + this.detectionRadius()
+               );
+      };
+
+      let dragging = false;
+      let interactionDragStartCallback = (start: Point) => {
+        if (onLine(start)) {
+          dragging = true;
+          this._dragStartCallbacks.callCallbacks(this);
+        }
+      };
+      this._dragInteraction.onDragStart(interactionDragStartCallback);
+
+      let interactionDragCallback = (start: Point, end: Point) => {
+        if (dragging) {
+          this._setPixelPositionWithoutChangingMode(this._isVertical() ? end.x : end.y);
+          this._dragCallbacks.callCallbacks(this);
+        }
+      };
+      this._dragInteraction.onDrag(interactionDragCallback);
+
+      let interactionDragEndCallback = (start: Point, end: Point) => {
+        if (dragging) {
+          dragging = false;
+          this._dragEndCallbacks.callCallbacks(this);
+        }
+      };
+      this._dragInteraction.onDragEnd(interactionDragEndCallback);
+
+      this._disconnectInteraction = () => {
+        this._dragInteraction.offDragStart(interactionDragStartCallback);
+        this._dragInteraction.offDrag(interactionDragCallback);
+        this._dragInteraction.offDragEnd(interactionDragEndCallback);
+        this._dragInteraction.detachFrom(this);
+      };
+
+      this._dragStartCallbacks = new Utils.CallbackSet<DragLineCallback<D>>();
+      this._dragCallbacks = new Utils.CallbackSet<DragLineCallback<D>>();
+      this._dragEndCallbacks = new Utils.CallbackSet<DragLineCallback<D>>();
+    }
+
+    protected _setup() {
+      super._setup();
+      this._detectionEdge = this.content().append("line").style({
+                              "opacity": 0,
+                              "stroke": "pink",
+                              "pointer-events": "visibleStroke"
```

Comment:
For consistency; `"pointer-events"` has to be quoted, in this case. Also matches the style used in `DragBoxLayer`.